main.c

```diff
@@ -70,6 +70,15 @@ typedef enum {
     INCREASE=1,
 } choice_event;
 
+choices_t *read_iso_choices(args_t *args)
+{
+    choices_t *choices = choices_create(args->num_infiles);
```

Comment:
So this is more a question about simple streams than anything else I think ... but wait a sec here.

As I understand it, this code chooses the most recent ISO from each simple stream file from a 2x2 matrix {daily,release}x{server,desktop}. Is that what we want? I'm not very convinced I'm afraid. Today this offers Lunar and Kinetic, in a few weeks it'll over Lunar and Mouldy, etc. We should only offer supported series by default and also should offer at least the most recent LTS.

So I'm going to be annoying and ask that we go back to the design phase a bit here (was this topic covered in a spec? I don't think so unfortunately).

The axes we should consider are {server, desktop} x {daily, release} x {stable ubuntu release} 

Right now of course, the only ubuntu version with release ISOs can actually install is Kinetic but 22.04.2 is close enough that we can assume that is out for these purposes. So already this is 8 options and eventually it might be 16 -- probably too many options for a single menu :/ But maybe we can wave that away for this PR.

Unfortunately, it doesn't look as if the simplestreams data actually indicates which releases are stable, so we get to parse https://changelogs.ubuntu.com/meta-release or something.

I don't really know how to indicate the distinction between a daily image and a release image. Maybe people who use this ISO understand this already... no way of knowing without getting it out into the world I suppose.

I don't know what this means for this PR. Maybe we should land it anyway and iterate?

Reply:
I strongly prefer landing what I have, and iterating.  The previous state is that the path to kinetic is hardcoded, lunar isn't much better, and no desktop handling.  

> was this topic covered in a spec?

No, but I'm not too worried.  I think we can pick a definition of what ISOs we want to show, and it should be small changes from here, except that I'm not handling multiple ISOs from one JSON source file yet and the handling of something being supported isn't in place yet.

> probably too many options for a single menu

I am assuming we will want some menu nesting at some point

> I don't really know how to indicate the distinction between a daily image and a release image.

Great question, it's in mind but I don't have an answer either.

Reply:
> I strongly prefer landing what I have, and iterating. The previous state is that the path to kinetic is hardcoded, lunar isn't much better, and no desktop handling.

OK, so long as we iterate before release :-)

> No, but I'm not too worried. I think we can pick a definition of what ISOs we want to show, and it should be small changes from here, except that I'm not handling multiple ISOs from one JSON source file yet and the handling of something being supported isn't in place yet.

Well OK. If it gets complex, having our thought processes in a google doc sounds sensible to me but I'm not going to insist.

> I am assuming we will want some menu nesting at some point

OK :-) I guess we get to decide the order the choices are presented...

>> I don't really know how to indicate the distinction between a daily image and a release image.

> Great question, it's in mind but I don't have an answer either.

Cloud images and lxd have the same issue of course. I wonder how they present it.